lib/improver/nbhood/square_kernel.py

```diff
@@ -611,12 +603,13 @@ def run(self, cube, radius, mask_cube=None):
             cubes_to_sum = (
                 self._set_up_cubes_to_be_neighbourhooded(cube_slice,
                                                          mask_cube))
-            neighbourhood_averaged_cubes = (
+            neighbourhood_averaged_cube = (
                 self._pad_and_calculate_neighbourhood(
                     cubes_to_sum, grid_cells_x, grid_cells_y))
             neighbourhood_averaged_cube = (
                 self._remove_padding_and_mask(
-                    neighbourhood_averaged_cubes, cubes_to_sum, cube.name(),
+                    neighbourhood_averaged_cube,
+                    cube_slice, mask_cube,
```

Comment:
What happens if the input cube has a mask. This gets used in the neighbourhooding, but not reapplied. I may have got lost going through this code and got it wrong.